web/controllers/task_controller.ex

```diff
@@ -57,30 +63,43 @@ defmodule CoursePlanner.TaskController do
   def edit(conn, %{"id" => id}) do
     case Tasks.get(id) do
       {:ok, task} ->
-        render(conn, "edit.html",
-          task: task,
-          changeset: Task.changeset(task),
-          users: Volunteers.all())
+        render(conn, "edit.html", task: task, changeset: Task.changeset(task))
       {:error, :not_found} ->
         conn
         |> put_status(404)
         |> render(CoursePlanner.ErrorView, "404.html")
     end
   end
 
-  def update(conn, %{"id" => id, "task" => params}) do
-    case Tasks.update(id, params) do
-      {:ok, task} ->
-        conn
-        |> put_flash(:info, "Task updated successfully.")
-        |> redirect(to: task_path(conn, :show, task))
-      {:error, :not_found} ->
-        conn
-        |> put_status(404)
-        |> render(CoursePlanner.ErrorView, "404.html")
-      {:error, task, changeset} ->
-        render(conn, "edit.html", task: task, changeset: changeset, users: Volunteers.all())
-    end
+  def update(conn, %{"id" => id, "task" => task_params})do
+    with {:ok, task} <- Tasks.get(id)
```

Comment:
Previously the updating logic was encapsulated into the `Tasks.update/2` function, I think that was an appropriate solution, so we could just adapt it to accept multiple volunteers instead of one